tf2onnx/optimizer/transpose_optimizer.py

```diff
@@ -0,0 +1,382 @@
+                    mul_val = numpy_helper.to_array(g.get_initializer_tensor(mul.input[1]))
+                    result = np.multiply(transposed_val, mul_val)
+                    g.update_initializer_tensor(conv.input[1], np.transpose(result, (3, 2, 0, 1)))
+
+                    ops = g.get_nodes()
+                    g.replace_subgraph_output(ops, p, trans)
+                    ops.remove(p)
+                    g.set_nodes(ops)
+                    return True
+                else:
+                    muler_dim = g.get_initializer_tensor(mul.input[1]).dims[0]
+                    if muler_dim == 1: # if there is only 1 number, so we just move trasponse after the mul
+                        ops = g.get_nodes()
+                        cnt = g.replace_subgraph_output(ops, p, trans)
+
+                        p.input[0] = trans.input[0]
+                        trans.input[0] = p.name + ":0"
+                        g.set_nodes(ops)
+                        return True
+                    else: # if the muler is not a single number, we need shift the data
+                        print("shift Conv's weight to NCHW....")
+            else:
+                print("Mul's second input is not a const, skipping")
+                pass
+        elif p.type == "Identity":
+            ops = g.get_nodes()
+            g.replace_subgraph_output(ops, p, trans)
+            ops.remove(p)
+            g.set_nodes(ops)
+            return True
+        elif p.type == "Concat":
+            if handle_node_having_branches(g, p) == True:
+                p.set_attr("axis", 1)
+                return True
+        elif p.type == "Split":
+            # Todo: need handle cases where Slit node has more than 1 outputs.
+            if handle_node_having_branches(g, p) == True:
+                p.set_attr("axis", 1)
+                return True
+
+        elif p.type == "Pad":
+            pad = p
+            #[N-start, H-start,  W-start, C-start, N-end, H-end,  W-end, C-end]
+            pads = pad.get_attr('pads').ints #[x1_begin, x2_begin...x1_end, x2_end,...]
+            # NHWC->NCHW
+            new_pads = [pads[0], pads[3], pads[1], pads[2], pads[4], pads[7], pads[5], pads[6] ]
+            p.set_attr("pads", new_pads)
+            switch_transpose_and_node(g, pad, trans)
+            return True
+
+        elif p.type == "ReduceMean":
+            axes = p.get_attr("axes").ints
+            keepdims = p.get_attr("keepdims")
+            # make sure keepdims is 1, then we can do the swap, otherwise, plese don't, because
+            # once keepdims is not set, original dims are lost, so transpose back won't work well.
+            # by default, if keepdims is not specified, it is 1
+            if axes == [1, 2] and ((keepdims and keepdims.i == 1) or (not keepdims)):
+                p.set_attr("axes", [2, 3])
+            else:
+                return
+            switch_transpose_and_node(g, p, trans)
+            return True
+

```

Comment:
how about ops like stridedslice ? I think you'd need to make a pass through all ops if any of the attributes is impacted.

Reply:
for ops that are not listed explicitly in the if branch, the transposes before it will remain there, so the correctness of the graph can be guaranteed. 

I'll add stridedslice support later.